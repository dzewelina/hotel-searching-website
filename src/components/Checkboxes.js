import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Checkboxes extends Component {
  state = {
    facilities: ['car park', 'restaurant', 'bar', 'gym', 'pool']
  };

  render() {
    const { facilities } = this.state;
    const { handleChange } = this.props;
    return (
      <div>
        <h2>Filters</h2>
        {facilities.map((facility, i) => (
          <div key={i} className='form-check'>
            <input className='form-check-input' type='checkbox' value={facility} onChange={handleChange} />
            <label className='form-check-label'>
              {facility}
            </label>
          </div>
        ))}
      </div>
    );
  }
}

Checkboxes.propTypes = {
  handleChange: PropTypes.func.isRequired
};

export default Checkboxes;