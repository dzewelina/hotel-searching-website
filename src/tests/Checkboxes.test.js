import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer';

import React from 'react';
import Checkboxes from '../components/Checkboxes';

describe('Checkboxes component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Checkboxes handleChange={jest.fn()}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});