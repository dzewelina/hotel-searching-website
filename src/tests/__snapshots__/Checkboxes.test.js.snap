// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkboxes component matches the snapshot 1`] = `
<div
  className="filters"
>
  <h2>
    Filters
  </h2>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
        value="car park"
      />
      car park
    </label>
  </div>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
        value="restaurant"
      />
      restaurant
    </label>
  </div>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
        value="bar"
      />
      bar
    </label>
  </div>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
        value="gym"
      />
      gym
    </label>
  </div>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        onChange={[Function]}
        type="checkbox"
        value="pool"
      />
      pool
    </label>
  </div>
</div>
`;
